services/state/token/account.proto

```diff
@@ -0,0 +1,176 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "basic_types.proto";
+import "crypto_approve_allowance.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * First-draft representation of a Hedera Token Service account entity in the network Merkle tree.
+ *
+ * As with all network entities, an account has a unique entity number, which is usually given along
+ * with the network's shard and realm in the form of a shard.realm.number id.
+ */
+
+message Account {
+    /**
+     * The unique entity number of the account.
+     */
+    int64 account_number = 1;
+    /**
+     * The public key bytes to be used as the account's alias, if any.
+     */
+    bytes alias = 2;
+    /**
+     * (Optional) The key to be used to sign transactions from the account, if any.
+     */
+    Key key = 3;
+    /**
+     * The expiration time of the account, in seconds since the epoch.
+     */
+    int64 expiry = 4;
+    /**
+     * The balance of the account, in tiny-bars.
+     */
+    int64 tinybar_balance = 5;
+    /**
+     * An optional description of the account with UTF-8 encoding up to 100 bytes.
+     */
+    string memo = 6;
+    /**
+     * A boolean marking if the account has been deleted.
+     */
+    bool deleted = 7;
+    /**
+     * A boolean marking if the account is a smart contract.
+     */
+    bool smart_contract = 8;
+    /**
+     * A boolean marking if the account requires a receiver signature.
+     */
+    bool receiver_sig_required = 9;
+    /**
+     * The number of NFTs owned by the account.
+     */
+    int64 number_owned_nfts = 10;
```

Comment:
Is there any reason we use `int64` here, but `int32` in associations? Should we use `int32` in all these places? I'm not sure if anybody is going to have more than 2 billion NFTs, but I guess it could happen... but maybe they should use multiple accounts at that point?

At the protobuf level int64 and int32 are just as efficient up until the limit of int32. But at the Java level it is a long vs. an int. Otherwise it wouldn't matter that much.

Reply:
Just used int64 where we were using long and `int32` for `int`. Will revisit before these going to production.